usecases/replica/replicator.go

```diff
@@ -33,20 +36,48 @@ type Replicator struct {
 	stateGetter shardingState
```

Comment:
Actually, I first had embedded Finder into Replicator, but to me it makes more sense to have finder as an attribute of the class Replicator.

This is because, when accessing the Finder for a NodeObject or FindOne, I think this is more clear:
```
r.finder.NodeObject(...)
r.finder.FindOne(...)
```
than this:
```
r.NodeObject(...)
r.FindOne(...)
```
because for the write handlers, we dont have:
```
r.PutObject(...)
```
but instead:
```
r.client.PutObject(...)
```
what do you think?

Reply:
I don't get the last point about r.client.PutObject(...).

The main reason is the finder and replicator interfaces will not overlap which results in more concise and less duplicated code.
Having said that I am also fine with the version you have suggested

Reply:
> I don't get the last point about r.client.PutObject(...)

I was just saying that the write methods are not directly accessed on the `Replicator` instance (have to go through `r.client`, so it seemed more consistent to put `r.finder` as the accessor for `GetObject`.

However, due to making the change you suggested [here](https://github.com/semi-technologies/weaviate/pull/2420#discussion_r1042435179), we won't have `GetObject` anymore, so this no longer matters 🙂 

Thanks for the explanation